src/pat/autotoc/autotoc.js

```diff
@@ -120,7 +130,49 @@ export default Base.extend({
                 skipHash: true,
             });
         }
+
+        // After DOM tree is built, initialize eventual validation
+        this.initialize_validation(self.$el);
+    },
+
+    initialize_validation: function ($el) {
+        const el = $el[0];
+
+        // Initialize only on pat-validation forms.
+        const form = el.closest("form.pat-validation");
+        if (! form) {
+            return;
+        }
+
+        for (const tab of this.tabs) {
+            if (tab.section.querySelectorAll("[required]").length > 0) {
+                tab.nav.classList.add("required");
+                tab.section.classList.add("required");
+            } else {
+                tab.nav.classList.remove("required");
+                tab.section.classList.remove("required");
+            }
+        }
+
+        form.addEventListener("pat-update", (e) => {
+            if (e.detail?.pattern !== "validation") {
+                // Nothing to do.
+                return;
+            }
+
+            for (const tab of this.tabs) {
+                if (tab.section.querySelectorAll("input:invalid").length > 0) {
```

Comment:
TODO: extend selector to also support textarea and select fields

Reply:
could this be generalized to:

```suggestion
                if (tab.section.querySelectorAll(":invalid").length > 0) {
```

Reply:
@petschki tnx, applied!